app/models/manageiq/providers/kubevirt/infra_manager/event_parser.rb

```diff
@@ -0,0 +1,25 @@
+class ManageIQ::Providers::Kubevirt::InfraManager::EventParser
+  def self.event_to_hash(event, ems_id = nil)
+    _log.debug("ems_id: [#{ems_id}] event: [#{event.inspect}]")
+
+    kind = event.object.involvedObject.kind
+
+    event_hash = {
+      :event_type          => "#{kind.upcase}_#{event.object.reason.upcase}",
+      :source              => 'KUBEVIRT',
+      :timestamp           => event.object.lastTimestamp,
+      :message             => event.object.message,
+      :container_namespace => event.object.involvedObject.namespace,
+      :full_data           => event.to_h,
```

Comment:
One thing to double check here, event is a RecursiveOpenStruct can you confirm that event.to_h converts the nested values to simple hash also and not ROS objects?

Reply:
Looks like it does
```
(byebug) pp event_hash
{:event_type=>"VIRTUALMACHINEINSTANCE_MIGRATED",
 :source=>"KUBEVIRT",
 :timestamp=>"2024-11-12T16:16:50Z",
 :message=>"EvictionStrategy is set but vmi is not migratable; cannot migrate VMI: PVC centos-stream9-scarlet-duck-44 is not shared, live migration requires that all PVCs must be shared (using ReadWriteMany access mode)",
 :container_namespace=>"miq",
 :full_data=>
  {:type=>"MODIFIED",
   :object=>
    {:kind=>"Event",
     :apiVersion=>"v1",
     :metadata=>
      {:name=>"centos-stream9-scarlet-duck-44.1807437ad1a2db12",
       :namespace=>"miq",
       :uid=>"cf6dc3a5-3223-4d86-b2cf-531fef74d25e",
       :resourceVersion=>"32920483",
       :creationTimestamp=>"2024-11-12T15:52:56Z",
       :managedFields=>
        [{:manager=>"virt-handler",
          :operation=>"Update",
          :apiVersion=>"v1",
          :time=>"2024-11-12T16:16:50Z",
          :fieldsType=>"FieldsV1",
          :fieldsV1=>{:"f:count"=>{}, :"f:firstTimestamp"=>{}, :"f:involvedObject"=>{}, :"f:lastTimestamp"=>{}, :"f:message"=>{}, :"f:reason"=>{}, :"f:reportingComponent"=>{}, :"f:reportingInstance"=>{}, :"f:source"=>{:"f:component"=>{}, :"f:host"=>{}}, :"f:type"=>{}}}]},
     :involvedObject=>{:kind=>"VirtualMachineInstance", :namespace=>"miq", :name=>"centos-stream9-scarlet-duck-44", :uid=>"b80fa3c9-c7c2-4c5c-9201-12ae7b6da1c8", :apiVersion=>"kubevirt.io/v1", :resourceVersion=>"32904613"},
     :reason=>"Migrated",
     :message=>"EvictionStrategy is set but vmi is not migratable; cannot migrate VMI: PVC centos-stream9-scarlet-duck-44 is not shared, live migration requires that all PVCs must be shared (using ReadWriteMany access mode)",
     :source=>{:component=>"virt-handler", :host=>"<worker>"},
     :firstTimestamp=>"2024-11-12T15:52:56Z",
     :lastTimestamp=>"2024-11-12T16:16:50Z",
     :count=>11,
     :type=>"Warning",
     :eventTime=>nil,
     :reportingComponent=>"virt-handler",
     :reportingInstance=>"<worker>"}},
 :ems_id=>"80",
 :ems_ref=>"cf6dc3a5-3223-4d86-b2cf-531fef74d25e",
 :vm_name=>"centos-stream9-scarlet-duck-44",
 :vm_ems_ref=>"b80fa3c9-c7c2-4c5c-9201-12ae7b6da1c8"}
 ```